### PR TITLE
Add more ways to connect to weaviate

### PR DESCRIPTION
**Pull request by ephraimbuddy**
There are other options for connecting to weaviate. This commit adds these other options and also improved the imports/typing

### CODE REVIEW

**Comment by utkarsharma2** [comment]
@ephraimbuddy It looks good to me, can we also update the relevant docs? - https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-weaviate/connections.rst

File: airflow/providers/weaviate/hooks/weaviate.py [hunk 1]
```diff
@@ -62,29 +63,35 @@ def get_ui_field_behaviour() -> dict[str, Any]:
             },
         }
 
-    def get_client(self) -> weaviate.Client:
+    def get_client(self) -> WeaviateClient:
```
Comment (utkarsharma2) [inline airflow/providers/weaviate/hooks/weaviate.py]: Also, should we rename this method to `get_conn`?

Comment (ephraimbuddy) [reply airflow/providers/weaviate/hooks/weaviate.py]: If we don't care for backward compatibility, I can rename this

Comment (utkarsharma2) [reply airflow/providers/weaviate/hooks/weaviate.py]: We can create another function `get_conn` which calls `get_client` and add a deprecation warning in `get_client`?

**Review by utkarsharma2 — APPROVED** [review]

File: airflow/providers/weaviate/hooks/weaviate.py (hunk 1 above)
Comment (ephraimbuddy) [reply airflow/providers/weaviate/hooks/weaviate.py]: There's already a `get_conn` in the PR. Yeah, good idea to add deprecation

**Review by phanikumv — APPROVED** [review]
LGTM

File: tests/providers/weaviate/hooks/test_weaviate.py [hunk 2]
```diff
@@ -16,12 +16,16 @@
...
 
+from unittest import mock
 from unittest.mock import MagicMock, Mock, patch
 
 import pytest
 
+from airflow.models import Connection
 from airflow.providers.weaviate.hooks.weaviate import WeaviateHook
 
+pytestmark = pytest.mark.db_test
```
Comment (ephraimbuddy) [inline tests/providers/weaviate/hooks/test_weaviate.py]:
```suggestion
```

**Pull request merged**